packages/documentation/src/content/docs/integration/deployment/helm-k8s.mdx

```diff
@@ -3,49 +3,874 @@ title: Helm & Kubernetes
+```
+
+#### Tigerbeetle backup
+
+Tigerbeetle is designed to be fault-tolerant with its replication mechanism. However, to create a backup of Tigerbeetle data, you can use the following approach:
+
+```bash
+# Create a snapshot of the Tigerbeetle PVC
+kubectl get pvc tigerbeetle-data-tigerbeetle-0 -o yaml > tigerbeetle-pvc.yaml
+
+# Create a volume snapshot
+cat <<EOF | kubectl apply -f -
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: tigerbeetle-snapshot
+spec:
+  volumeSnapshotClassName: csi-hostpath-snapclass
+  source:
+    persistentVolumeClaimName: tigerbeetle-data-tigerbeetle-0
+EOF
+```
+
+:::note
+The above example assumes you have a <LinkOut href='https://kubernetes.io/docs/concepts/storage/volumes/#csi'>Container Storage Interface (CSI)</LinkOut> driver capable of volume snapshots. Adjust the `volumeSnapshotClassName` according to your cluster setup.
 :::
+
+### Database recovery
+
+#### PostgreSQL recovery
+
+To restore from a PostgreSQL backup:
+
+```bash
+# Forward PostgreSQL port to local machine
+kubectl port-forward svc/rafiki-postgresql 5432:5432
+
+# Use psql to restore from backup
+psql -h localhost -U rafiki -d rafiki < rafiki_pg_backup.sql
+```
+
+#### Tigerbeetle recovery
+
+To restore Tigerbeetle from a snapshot:
+
+```bash wrap
+# Create a new PVC from the snapshot
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: tigerbeetle-data-restored
+spec:
+  dataSource:
+    name: tigerbeetle-snapshot
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+EOF
+
+# Update the Tigerbeetle StatefulSet to use the restored PVC
+kubectl patch statefulset tigerbeetle -p '{"spec":{"template":{"spec":{"volumes":[{"name":"data","persistentVolumeClaim":{"claimName":"tigerbeetle-data-restored"}}]}}}}'
+```
+
+## High availability configuration
+
+For production environments requiring high availability:
+
+```yaml
+components:
+  authServer:
+    replicaCount: 3
+    podDisruptionBudget:
+      enabled: true
+      minAvailable: 2
+
+  backendApi:
+    replicaCount: 3
+    podDisruptionBudget:
+      enabled: true
+      minAvailable: 2
+
+  engineServer:
+    replicaCount: 3
+    podDisruptionBudget:
+      enabled: true
+      minAvailable: 2
+
+  # Configure other components similarly
+
+postgresql:
+  architecture: replication
+  replication:
+    enabled: true
+    readReplicas: 2
+
+redis:
+  cluster:
+    enabled: true
+    slaveCount: 2
+
+tigerbeetle:
+  replicaCount: 3 # Use multiple replicas for fault tolerance
+  persistence:
+    enabled: true
+    size: 20Gi # Larger storage for production
+  resources:
+    limits:
+      cpu: 1000m
+      memory: 2Gi
+    requests:
+      cpu: 500m
+      memory: 1Gi
+```
+
+This configuration sets up multiple replicas for each Rafiki component and enables replication for both PostgreSQL, Redis, and Tigerbeetle.
```

Comment:
The last suggestion I have is to move this line of text so it appears before the code block.

> For production environments requiring high availability, the following configuration ....

That's a really long sentence though.